url.js

```diff
@@ -76,9 +76,12 @@
     var instance = URLUtils(url || '');
 
     // Detect for ES5 getter/setter support
-    var ES5_GET_SET = (Object.defineProperties && (function () {
-      var o = {}; Object.defineProperties(o, { p: { 'get': function () { return true; } } }); return o.p;
-    }()));
+    var ES5_GET_SET = false;
+    try {
+      ES5_GET_SET = (Object.defineProperties && (function () {
```

Comment:
Can you add a short comment explaining why the try/catch is needed? e.g.

// An Object.defineProperties polyfill that doesn't support getters/setters may throw.